Program.cs

```diff
@@ -42,39 +29,14 @@
 			}
 		}
 	}
-	else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) {
-		ELF<ulong> elf;
-		if (!ELFReader.TryLoad(new MemoryStream(moduleBytes), true, out elf)) {
-			throw new ArgumentException($"{Path.GetFileName(modulePath)} is not a linux executable.");
-		}
-
-		var text = (ProgBitsSection<ulong>) elf.GetSection(".text");
-		var symOffset = text.Offset - text.LoadAddress;
-
-		var sym = (SymbolTable<ulong>) elf.GetSection(".symtab");
-
-		foreach (var patch in Patches.Linux) {
-			var func = sym.Entries.Single(e => e.Name.Equals(patch.FunctionName));
-			var fnBytes = moduleBytes.AsSpan((int)(func.Value + symOffset), (int)func.Size);
-			if (patch.Apply(fnBytes)) {
-				Console.WriteLine($"Patched {patch.FunctionName}");
-				applied = true;
-			} else {
-				Console.WriteLine($"Already patched {patch.FunctionName}");
-			}
-		}
-	}
-	else {
-		throw new NotSupportedException($"Unsupported platform: {RuntimeInformation.OSDescription}");
-	}
 
 	if (applied)
 		File.WriteAllBytes(modulePath, moduleBytes);
 
 	Console.WriteLine("Done");
 }
 catch (ArgumentException ex) {
-	Console.Error.WriteLine(ex.Message);
```

Comment:
This was intentional. Otherwise there's no point separating out `ArgumentException`. I used it to provide user facing error messages